### DISCUSSION TITLE
Livewire Component disappears when re-rendering, If I put an HTML comment on Line Number 1

### DISCUSSION THREAD

**Discussion by alinauf** [livewire/livewire#4101] [1. Help] [answered]
When I put an HTML comment at the beginning of the file on livewire component and if I try to re-render the page the whole component disappears. I'm not sure if it is just me but this happens on every component I've tested this on.

**Livewire Class**
```

class Show extends Component
{
    use WithPagination;
    public $search;

    public function render()
    {

        $data= Example::where('name', 'like', '%' . $this->search . '%')
            ->orderBy('id', 'desc')
            ->get();

        return view('livewire.admin.transaction.invoice.show',['examples'=>$data]);
    }
}

```

**Livewire Component:**
```
<!-- This HTML Comment on Line number 1 causes the issue-->
<div>
    <input type="search" wire:model="search" id="search"
           class="mt-1 block w-full border border-gray-300 rounded-md shadow-sm py-2 px-3 focus:outline-none focus:ring-primary-500 focus:border-primary-500 sm:text-sm">

    @foreach($examples as $example )
        <div>
            {{$example->name}}
        </div>
    @endforeach

</div>

```

**Comment by joshhanley** [comment, marked as answer]
@alinauf Livewire only supports having one root html element, see here in the docs https://laravel-livewire.com/docs/2.x/rendering-components#returning-blade

> Make sure your Blade view only has ONE root element.

But html comments are considered as a valid html element, as such you should put the comment inside the div or consider changing it to a blade comment `{{-- comment --}}`

Hope this helps!

**Reply by alinauf** [reply]
Ah alright. I never knew the comment would be considered as an element. Thanks!

**Reply by joshhanley** [reply]
No worries! 🙂
